File: client/neutroview/src/app/model/article_query/article-query.ts
export class ArticleQuery {
  keyword?: string;
  category?: string;
  timespan?: string;
  numRecords?: number;
  domain?: string;
  startDate?: Date;
  endDate?: Date;
  country?: string;

  constructor(
    keyword?: string,
    category?: string,
    timespan?: string,
    numRecords?: number,
    domain?: string,
    startDate?: Date,
    endDate?: Date,
    country?: string
  ) {
    this.keyword = keyword;
    this.category = category;
    this.timespan = timespan;
    this.numRecords = numRecords;
    this.domain = domain;
    this.startDate = startDate;
    this.endDate = endDate;
    this.country = country;
  }
}

File: server/model/article_query.py
from dataclasses import dataclass
from datetime import date
from typing import Optional


@dataclass
class ArticleQuery:
    keyword: Optional[str] = None
    category: Optional[str] = "politics"
    timespan: Optional[str] = "1d"
    num_records: Optional[int] = 10
    domain: Optional[str] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    country: Optional[str] = None
